pkg/lib/serverTime.js

```diff
@@ -321,13 +309,102 @@
+
+    function get_custom_ntp_chronyd() {
+        const enabled_file = cockpit.file(chronyd_sources_enabled, { superuser: "try" });
+        const disabled_file = cockpit.file(chronyd_sources_disabled, { superuser: "try" });
+
+        function parse_servers(data) {
+            if (!data)
+                return [];
+            const servers = [];
+            data.split("\n").forEach(function(line) {
+                const parts = line.split(" ");
+                if (parts[0] == "server")
+                    servers.push(parts[1]);
+            });
+            return servers;
+        }
+
+        return enabled_file.read()
+                .then(data => {
+                    if (data) {
+                        return {
+                            backend: "chronyd",
+                            enabled: true,
+                            servers: parse_servers(data)
+                        };
+                    } else {
+                        return disabled_file.read()
+                                .then(data => {
+                                    return {
+                                        backend: "chronyd",
+                                        enabled: false,
+                                        servers: parse_servers(data)
+                                    };
+                                });
+                    }
+                });
+    }
+
+    function set_custom_ntp_chronyd(config) {
+        const enabled_file = cockpit.file(chronyd_sources_enabled, { superuser: "try" });
+        const disabled_file = cockpit.file(chronyd_sources_disabled, { superuser: "try" });
+
+        const text = "# This file is automatically generated by Cockpit\n\n" + config.servers.map(s => `server ${s}\n`).join("");
+
+        function ensure_sourcedir() {
+            function add_sourcedir(data) {
+                const line = "sourcedir " + chronyd_sourcedir;
+                if (data && data.indexOf(line) == -1)
+                    data += "\n# Added by Cockpit\n" + line + "\n";
+                return data;
+            }
+            return cockpit.file("/etc/chrony.conf", { superuser: "try" }).modify(add_sourcedir);
+        }
+
+        return cockpit.spawn(["mkdir", "-p", chronyd_sourcedir], { superuser: "try" })
+                .then(() => {
+                    if (config.enabled)
+                        return enabled_file.replace(text).then(() => disabled_file.replace(null)).then(ensure_sourcedir);
+                    else
+                        return disabled_file.replace(text).then(() => enabled_file.replace(null));
+                });
+    }
+
+    self.get_custom_ntp = function () {
+        return get_timesync_backend().then(backend => {
+            if (backend == "timesyncd")
+                return get_custom_ntp_timesyncd();
+            else if (backend == "chronyd")
+                return get_custom_ntp_chronyd();
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18298-20230210-080149-0a3aba95-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)